### ISSUE TITLE
Should we throw an error when fetch is aborted?

### ISSUE THREAD

**Issue by Liroo** [dai-shi/react-hooks-async#30]
Do we really need to throw an error without stacktrace when a fetch request is aborted ?

We could check the `Error` name as:

```
try {
    result = await func(task.abortController, ...args);
} catch (e) {
   if (e.name !== 'AbortError') {
      err = e;
   }
}
```

in order to reduce the number of error in the console. Furthermore it's not an error, it's a wanted behavior that we want to abort the error with a signal.

What do you think?

**Comment by dai-shi** [comment]
Thanks for the suggestion!
I think it's reasonable. I don't think there are edge cases that would be broken with it, because only the lib can abort fetch.
One little question is whether we should implement that in useAsyncTask or useAsyncTaskFetch.
Yeah, but let's make it a convention. AbortController and AbortError should be paired.
@Liroo Would you like to draft a PR?

**Comment by Liroo** [comment]
I will make a PR when I get time next days!

Mhh.. I mean, it's the usAsyncTask that generate the AbortController, but only the useAsyncTaskfetch can use it.. what to do..

**Comment by dai-shi** [comment]
Cool. I think it's fine to implement in useAsyncTask. It makes more sense. That means if useAsyncTaskAxios throws an error on abort, it should be AbortError.

**Comment by dai-shi** [comment]
> useAsyncTaskAxios throws an error on abort, it should be AbortError.

Fixed in 08519defcc624e5741f3a67625cfcc44c78b788e
